types/chain_info.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// ChainInfo is used to track metadata about a peer and its chain.
+type ChainInfo struct {
+	Peer   peer.ID
+	Head   TipSetKey
+	Height uint64
+}
+
+// NewChainInfo creates a chain info from a peer id a head tipset key and a
+// chain height.
+func NewChainInfo(peer peer.ID, head TipSetKey, height uint64) *ChainInfo {
+	return &ChainInfo{
+		Peer:   peer,
+		Head:   head,
+		Height: height,
+	}
+}
+
+// CISlice is for sorting chain infos
+type CISlice []*ChainInfo
+
+// Len returns the number of chain infos in the slice.
+func (cis CISlice) Len() int { return len(cis) }
+
+// Swap swaps chain infos.
+func (cis CISlice) Swap(i, j int) { cis[i], cis[j] = cis[j], cis[i] }
+
+// Less compares chain infos on peer ID.  There should only ever be one chain
+// info per peer in a CISlice.
+func (cis CISlice) Less(i, j int) bool { return string(cis[i].Peer) < string(cis[j].Peer) }
```

Comment:
If we only added these for testing I would prefer to find a different way to make the assertion.

Reply:
Is your thinking that this is unnecessary for such small test code?  I can see that but argue that it is not much code, properly isolated, a standard go / PL pattern (I lifted it from peer ids), and likely to be used in tests (maybe logging) down the line.